docs/Technical-Documentation/jwt-key-rotation.md

```diff
@@ -0,0 +1,70 @@
+# Rotating JWT Keys
+
+## Context
+
+To maintain good security, we will periodically rotate our JWT keys. This document outlines the process of how to do this.
+
+**Warning** Production sites will need to be taken down for maintenance when rotating keys, as the rotation will automatically invalidate all current sessions.
+
+### 1. Generate New Keys
+
+In your Mac terminal (or bash terminal in Windows), enter the following command:
+```bash=
+ssh-keygen -t rsa -b 4096 -m PEM -f jwtRS256.key
+```
+You will receive the following response. Do not add a passphrase when prompted.
+```
+Enter passphrase (empty for no passphrase):
+Enter same passphrase again:
+Your identification has been saved in jwtRS256.key
+Your public key has been saved in jwtRS256.key.pub
+The key fingerprint is:
+# returns key fingerprint
+The key's randmart image is:
+# returns key's randmart image
+```
+Then enter
+```bash=
+openssl rsa -in jwtRS256.key -pubout -outform PEM -out jwtRS256.key.pub
+```
+
+You can now check the contents of your keys with these commands
+```bash=
+cat jwtRS256.key
+# returns private key
+cat jwtRS256.key.pub
+# returns public key
+```
+
+### 2. Base64 Encode Private Key
+
+We use Base64 Encoded Private Keys to make it easier to save to cloud environments and local `.env` files.
+
+```bash=
+openssl enc -base64 -in jwtRS256.key -out jwtRS256.key.base64
+cat jwtRS256.key.base64
+```
+
+### 3. Copy Keys
+
+#### Dev/Staging Environments
+1. Distribute the private key to development staff securely to copy to `.env` files as the value for key `JWT_KEY`
+2. Update the variable `JWT_KEY` with the private key in cloud.gov backend development and staging environments
+3. Login to the [Login.gov Sandbox](https://dashboard.int.identitysandbox.gov/) and update the public key there
```

Comment:
Would it be safe to put a screenshot of where to go in their UI here? As long as nothing sensitive is shown I think that would be helpful to have here.

Reply:
Also, does Login.gov expect the key in PEM format or Base 64?

Reply:
PEM format. I have updated the doc with a note and a screenshot